### PR TITLE
Convert results with to_native in consul_kv plugin

### PR DESCRIPTION
**Pull request by MatrixCrawler**
##### SUMMARY
 Fix for #42851

Convert results with to_native

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lookup/consul_kv

##### ANSIBLE VERSION
```
ansible 2.6.4
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.6 (default, Nov 23 2017, 15:49:48) [GCC 4.8.4]
```

### CODE REVIEW

**Comment by neekz0r** [comment]
Thanks. This bug just bit us in a very uncomfortable location. :-)

**Comment by MatrixCrawler** [comment]
hm? What is the poblem with the tests? I don't get it

**Comment by s-hertel** [comment]
@MatrixCrawler rerunning them since they look unrelated

**Comment by MatrixCrawler** [comment]
I just realized that i branched from devel. Was that smart for fixing the bug?
Shouldn't i have branched from 2.6?

**Comment by s-hertel** [comment]
@MatrixCrawler Branching from devel is right. Merge to devel first and then you can cherry-pick the commit to the 2.7 and 2.6 stable branches. There's a document for the process here: https://docs.ansible.com/ansible/latest/community/development_process.html#backport-pull-request-process

Oh - but you've added metadata, and I don't think lookups handle that yet.

**Comment by MatrixCrawler** [comment]
Thanks for the info. Already did this once. Was just unsure for a short moment.

> Oh - but you've added metadata, and I don't think lookups handle that yet

@s-hertel i was asked by @gundalow in irc  to just add it regardless that it is not currently interpreted.

File: lib/ansible/plugins/lookup/consul_kv.py [hunk 1]
```diff
@@ -5,6 +5,10 @@
 
 __metaclass__ = type
 
+ANSIBLE_METADATA = {'metadata_version': '1.1',
```
Comment (Akasurde) [inline lib/ansible/plugins/lookup/consul_kv.py:8]: Not sure if this is required or not in plugin. cc @bcoca

Comment (MatrixCrawler) [reply lib/ansible/plugins/lookup/consul_kv.py:8]: I asked that in IRC yesterday and @gundalow asked me to add it as they are planning to implement the check for ansible metadata for plugins, too.

Comment (Akasurde) [reply lib/ansible/plugins/lookup/consul_kv.py:8]: I am sorry, I was not aware of this thing in pipeline.

File: lib/ansible/plugins/lookup/consul_kv.py [hunk 2]
```diff
@@ -99,8 +103,7 @@
 from ansible.module_utils.six.moves.urllib.parse import urlparse
 from ansible.errors import AnsibleError, AnsibleAssertionError
 from ansible.plugins.lookup import LookupBase
-
-import json
+from ansible.module_utils._text import to_native
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/consul_kv.py]: I suspect you actually want 'to_text' as bytestrings are not going to be very usable in jinja2 context

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/lookup/consul_kv.py (hunk 2 above)
Comment (MatrixCrawler) [reply lib/ansible/plugins/lookup/consul_kv.py]: Right you are. Fixed it

**Review by bcoca — APPROVED** [review]

**Pull request merged**